tools/validators/ontology_validator/yamlformat/validator/entity_type_lib.py

```diff
@@ -525,6 +525,32 @@ def GetAllFields(self, run_unsafe=False):
       self._all_fields = tmp
     return self._all_fields
 
+
+  def HasField(self, fully_qualified_fieldname, run_unsafe=False):
+    """Returns a boolean if a field is present or not
+    
+    Args:
+      fully_qualified_fieldname: a fully qualified names for example:
+      "HVAC/run_status_1.
+      run_unsafe: set true to run against a type before fields are fully
+        expanded.  Running in this mode does not memoize the result.
+    Returns:
+      true if the field is present, false otherwise.
+    """
+    fqf_parsed = fully_qualified_fieldname.split('/')
+    if len(fqf_parsed) == 1:
```

Comment:
these are effectively runtime errors.  You can just raise them.

Reply:
It's ok to raise these because they should only if there's a bug in the caller's code.